mod_wsbridge/mod_wsbridge.c

```diff
@@ -384,6 +450,88 @@ websocket_write_back(struct lws *wsi_in, enum lws_write_protocol type, char *str
 	return n;
 }
 
+void add_content_type(private_t *tech_pvt, cJSON* json) {
+	if (cJSON_GetObjectItem(json, "content-type")) {
+		cJSON_DeleteItemFromObject(json, "content-type");
+	}
+	cJSON_AddItemToObject(json, "content-type", cJSON_CreateString(tech_pvt->content_type));
+}
+
+int is_mute_event(char* event, char* method) {
+	return !strcmp(event, "websocket:media:update") && !strcmp(method, "update");
+}
+
+void reset_ws_write_indexes(private_t *tech_pvt) {
+	switch_mutex_lock(tech_pvt->write_mutex);
+	tech_pvt->write_count = 0;
+	tech_pvt->write_start = 0;
+	tech_pvt->write_end = 0;
+	switch_mutex_unlock(tech_pvt->write_mutex);
+}
+
+char* get_valuestring(cJSON* json, char* field) {
+	char* valuestring = NULL;
+	cJSON* json_field = NULL;
+	json_field = cJSON_GetObjectItem(json, field);
+	if (json_field) {
+		valuestring = json_field->valuestring;
+	}
+	return valuestring;
+}
+
+void on_event(private_t *tech_pvt, cJSON* json) {
+	char* event = NULL;
+	char* method = NULL;
+	event = get_valuestring(json, "event");
+	method =  get_valuestring(json, "method");
+
+	if (!event || !method) {
+		return;
+	}
+
+	if (is_mute_event(event, method)) {
+		cJSON* active;
+		active = cJSON_GetObjectItem(json, "active");
+		if (active && (active->type == cJSON_False || active->type == cJSON_True)) {
+			AudioActive_set(&tech_pvt->audio_active, active->valueint);
+			if(!active->valueint) {
+				reset_ws_write_indexes(tech_pvt);
+			}
+		}
+	}
+}
+
+void send_json_message(struct lws *wsi, cJSON* json_message) {
+	char *message = NULL;
+	if ((message = cJSON_PrintUnformatted(json_message)) != NULL) {
```

Comment:
You could use a precomputed buffer by the size of `size + 2` pass in the second element as the begging of the buffer and like that you won't need to reallocate for the 2 extra chars.

Reply:
used `cJSON_PrintPreallocated` with preallocated buffer of `size + 2`